lib/utils/jsonPathUtils.ts

```diff
@@ -90,4 +90,45 @@ export class JsonPathUtils {
       }
     }
   }
+
+  static changeSpecialPathsRecursively(pd: IPresentationDefinition) {
+    const paths: { value: unknown; path: (string | number)[] }[] = JsonPathUtils.extractInputField(pd, ['$..path']);
+    for (const path of paths) {
+      this.modifyPathsWithSpecialCharacter(path.path, pd);
+    }
+  }
+
+  private static modifyPathsWithSpecialCharacter(pathDetails: (string | number)[], pd: IPresentationDefinition) {
+    let objectCursor: any = pd;
+    for (let i = 1; i < pathDetails.length; i++) {
+      if (i + 1 < pathDetails.length) {
+        objectCursor = objectCursor[pathDetails[i]];
+      }
+      if (pathDetails.length == i + 1) {
+        const paths: string[] = objectCursor[pathDetails[i]];
+        const editedPaths: string[] = [];
+        for (let j = 0; j < paths.length; j++) {
+          editedPaths.push(this.modifyPathWithSpecialCharacter(paths[j]));
+        }
+        objectCursor[pathDetails[i]] = editedPaths;
+        break;
+      }
+    }
+  }
+
+  private static modifyPathWithSpecialCharacter(path: string): string {
+    const REGEX_PATH = /(^((?!\[').)*)(@\w+)/g;
+    const matches = path.matchAll(REGEX_PATH);
+    let nextExist = true;
+    while (nextExist) {
+      const next = matches.next();
+      if (!next.done) {
+        path = path.replace(next.value[3], ".['" + next.value[3] + "']");
```

Comment:
Why are we adding dots?

I either want a direct JSONLD term in my path by using a single dot or it needs to be somewhere in the path by using two dots. Why are we adding dots during our escaping? Please note that the improved code I posted above does the same to not break current tests.